common/src/main/java/com/wynntils/core/features/Feature.java

```diff
@@ -34,119 +19,19 @@
-
-                assert !overlay.getTranslatedName().startsWith("feature.wynntils.");
-            } catch (IllegalAccessException e) {
-                WynntilsMod.error("Unable to get field " + overlayField, e);
-            }
-        }
-    }
-
-    public final void initOverlayGroups() {
-        // Unregister old instances
-        for (Overlay groupedOverlay : groupedOverlayInstances) {
-            Managers.Overlay.unregisterOverlay(groupedOverlay);
-        }
-
-        groupedOverlayInstances.clear();
-
-        // Go on with discovering the new ones
-        Field[] groupFields = FieldUtils.getFieldsWithAnnotation(this.getClass(), OverlayGroup.class);
-
-        for (Field groupField : groupFields) {
-            try {
-                Object fieldValue = FieldUtils.readField(groupField, this, true);
-                OverlayGroup annotation = groupField.getAnnotation(OverlayGroup.class);
-
-                if (!(fieldValue instanceof List<?> list)) {
-                    throw new RuntimeException("A non overlay group field was marked with OverlayGroup annotation.");
-                }
-
-                for (Overlay overlay : (List<Overlay>) list) {
-                    Managers.Overlay.registerOverlay(overlay, annotation.renderType(), annotation.renderAt(), this);
-                    groupedOverlayInstances.add(overlay);
-
-                    assert !overlay.getTranslatedName().startsWith("feature.wynntils.");
-                }
-            } catch (IllegalAccessException e) {
-                WynntilsMod.error("Unable to get field " + groupField, e);
-            }
-        }
-    }
-
-    public final void addOverlayGroups(List<OverlayGroupHolder> groups) {
-        overlayGroups.addAll(groups);
-    }
-
-    public List<OverlayGroupHolder> getOverlayGroups() {
-        return overlayGroups;
-    }
-
-    /**
-     * Sets up this feature as an event listener. Called from the registry.
-     */
-    public final void setupEventListener() {
-        this.isListener = true;
-    }
-
-    /**
-     * Adds a keyBind to the feature. Called from the registry.
-     * @param keyBind KeyBind to add to the feature
-     */
-    public final void setupKeyHolder(KeyBind keyBind) {
-        keyBinds.add(keyBind);
-    }
-
-    public List<Overlay> getOverlays() {
-        return Stream.concat(overlays.keySet().stream(), groupedOverlayInstances.stream())
-                .toList();
+    public Category getCategory() {
+        return category;
     }
 
-    public OverlayInfo getOverlayInfo(Overlay overlay) {
-        return overlays.get(overlay);
+    public void setCategory(Category category) {
+        this.category = category;
     }
 
-    public final void enableOverlays() {
-        Managers.Overlay.enableOverlays(this.getOverlays(), false);
+    /** Whether a feature is enabled */
+    public final boolean isEnabled() {
+        return Managers.Feature.getFeatureState(this) == FeatureState.ENABLED;
```

Comment:
I'm thinking we should even have `Managers.Feature.isEnabled(this)`. If a feature is enabled is perhaps not as easy as looking at a state, I guess it can be "user enabled" but "crashed" which will in effect make it "disabled". Or did we chose to encode crashness in the state? In any case, I think a `isEnabled` would be nice and clear.

Reply:
The state existed long before the refactor, it was moved. I will add the method.